File: LeetCode_2300/Main.py
class Solution(object):
    def good(self, spell, potions, success):
        ans = len(potions)
        lo = 0
        hi = len(potions) - 1
    
        while (lo <= hi):
            mid = (lo + hi) // 2
    
            if (spell * potions[mid] >= success):
                ans = mid
                hi = mid - 1
            else:
                lo = mid + 1
        return ans
    def successfulPairs(self, spells, potions, success):
        """
        :type spells: List[int]
        :type potions: List[int]
        :type success: int
        :rtype: List[int]
        """
        answer = []
        n = len(potions)
        potions.sort()
        for spell in spells:
            answer.append(n - self.good(spell, potions, success))
        
        return answer

File: LeetCode_2300/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int good(int spell, vector<int>& potions, long long success) {
        int ans = potions.size();
        int lo = 0, hi = potions.size() - 1;
    
        while (lo <= hi) {
            int mid = (lo + hi) / 2;
    
            if (1LL * spell * potions[mid] >= success) {
                ans = mid;
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return ans;
    }

    vector<int> successfulPairs(vector<int>& spells, vector<int>& potions, long long success) {
        vector<int> answer;
        int n = potions.size();
        sort(potions.begin(), potions.end());

        for (int spell : spells) {
            answer.push_back(n - good(spell, potions, success));
        }

        return answer;
    }
};